library/DataDefs.cpp

```diff
@@ -159,6 +159,14 @@ enum_identity::enum_identity(size_t size,
     }
 }
 
+enum_identity::enum_identity(enum_identity *base_enum, type_identity *override_base_type)
+    : enum_identity(override_base_type->byte_size(), base_enum->getScopeParent(),
+                    base_enum->getName(), override_base_type, base_enum->first_item_value,
+                    base_enum->last_item_value, base_enum->keys, base_enum->complex,
+                    base_enum->attrs, base_enum->attr_type)
+{
+}
```

Comment:
Nice! This actually appears to work for fields even outside of vectors now, e.g. you can modify any fields in this structure from `gui/gm-editor` and see enum metadata:
```xml
        <enum name='season8' type-name='season' base-type='int8_t'/>
        <enum name='season16' type-name='season' base-type='int16_t'/>
        <enum name='season32' type-name='season' base-type='int32_t'/>
        <stl-vector name='v_season8'>
            <enum type-name='season' base-type='int8_t'/>
        </stl-vector>
        <stl-vector name='v_season16'>
            <enum type-name='season' base-type='int16_t'/>
        </stl-vector>
        <stl-vector name='v_season32'>
            <enum type-name='season' base-type='int32_t'/>
        </stl-vector>
```

Was there a reason for bundling this with this PR? If we're confident in it, I can see it being useful as a standalone change.

(One oddity on my end is that inserting an element into the 16-bit and 32-bit vectors in gui/gm-editor with `Alt-i`, `Enter` seems to produce uninitialized elements, but not the 8-bit vector. Not sure if that's an issue somewhere else, though.)

Reply:
This can be cherry-picked if desired. The reason it's here is that I was working on this:

https://github.com/DFHack/df-structures/blob/9b7ee3879aa07fe9f0348052db97e62a23d05863/df.history.xml#L543-L556